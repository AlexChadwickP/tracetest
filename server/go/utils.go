package openapi

import (
	"net/http"

	"github.com/gogo/protobuf/jsonpb"
	"github.com/gogo/protobuf/proto"
	v11 "go.opentelemetry.io/proto/otlp/common/v1"
	res "go.opentelemetry.io/proto/otlp/resource/v1"
	v1 "go.opentelemetry.io/proto/otlp/trace/v1"
)

func EncodeJSONPBResponse(i interface{}, status *int, w http.ResponseWriter) error {
	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
	if status != nil {
		w.WriteHeader(*status)
	} else {
		w.WriteHeader(http.StatusOK)
	}

	trace := i.(proto.Message)
	m := jsonpb.Marshaler{}
	return m.Marshal(w, trace)
}

func FixParent(tr *v1.TracesData, traceID, parentSpanID string) *v1.TracesData {
	spans := make(map[string]*v1.Span)
	for _, rs := range tr.ResourceSpans {
		for _, ils := range rs.InstrumentationLibrarySpans {
			for _, sp := range ils.Spans {
				spans[string(sp.SpanId)] = sp
			}
		}
	}

	// Fix parent id
	for _, sp := range spans {
		if sp.ParentSpanId == nil {
			continue
		}
		_, ok := spans[string(sp.ParentSpanId)]
		if !ok {
			sp.ParentSpanId = []byte(parentSpanID)
		}
	}
	rs := &v1.ResourceSpans{
		Resource: &res.Resource{
			Attributes:             []*v11.KeyValue{},
			DroppedAttributesCount: 0,
		},
		InstrumentationLibrarySpans: []*v1.InstrumentationLibrarySpans{
			{
				InstrumentationLibrary: &v11.InstrumentationLibrary{
					Name:    "tracetest",
					Version: "v1",
				},
				Spans: []*v1.Span{
					{TraceId: []byte(traceID),
						SpanId:       []byte(parentSpanID),
						ParentSpanId: nil,
						Name:         "tracetest",
						Kind:         v1.Span_SPAN_KIND_CLIENT,
					},
				},
			},
		},
		SchemaUrl: "",
	}
	tr.ResourceSpans = append(tr.ResourceSpans, rs)

	return tr
}
