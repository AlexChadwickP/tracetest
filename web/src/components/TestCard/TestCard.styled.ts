import {MoreOutlined} from '@ant-design/icons';
import {Typography} from 'antd';
import styled from 'styled-components';

export const TestCard = styled.div`
  display: grid;
  align-items: center;
  grid-template-columns: 20px 1fr 60px 2fr 220px 100px 20px;
  gap: 24px;
  padding: 24px;
  box-shadow: 0px 4px 8px rgba(153, 155, 168, 0.1);
  background: #fff;
  cursor: pointer;
`;

export const TextContainer = styled.div`
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const NameText = styled(Typography.Text)`
  font-weight: 700;
  overflow-x: ellipsis;
`;

export const Text = styled(Typography.Text)``;

export const ActionButton = styled(MoreOutlined).attrs({
  style: {fontSize: 24, color: '#9AA3AB', cursor: 'pointer'},
})``;
