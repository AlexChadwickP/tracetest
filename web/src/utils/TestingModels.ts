import {ITest} from '../types/Test.types';
import {IAssertionResult} from '../types/Assertion.types';
import {HTTP_METHOD} from '../constants/Common.constants';
import {ITestRunResult} from '../types/TestRunResult.types';
import {TestState} from '../constants/TestRunResult.constants';

interface IProps {
  test: ITest;
  assertionResult: IAssertionResult;
  testRunResult: ITestRunResult;
}

const test: ITest = {
  assertions: [],
  description: '',
  lastTestResult: undefined,
  name: '',
  serviceUnderTest: {
    id: '',
    request: {
      auth: undefined,
      body: '',
      certificate: undefined,
      headers: undefined,
      method: HTTP_METHOD.GET,
      proxy: undefined,
      url: '',
    },
  },
  testId: '',
};

const assertionResult: IAssertionResult = {
  assertion: {assertionId: '', selectors: [], spanAssertions: []},
  spanListAssertionResult: [],
};
const testRunResult: ITestRunResult = {
  assertionResult: [{assertionId: '', spanAssertionResults: []}],
  assertionResultState: false,
  completedAt: '',
  createdAt: new Date().toISOString().toString(),
  response: undefined,
  resultId: '',
  spanId: '',
  state: TestState.CREATED,
  testId: '',
  trace: undefined,
  traceId: '',
};

export const TestingModels: IProps = {
  assertionResult,
  test,
  testRunResult,
};
