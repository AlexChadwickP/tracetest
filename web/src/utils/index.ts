const flattenAttributesSelector = () =>
  `resourceSpans[].[instrumentationLibrarySpans[].spans[].attributes[].{key:key,value:value.*|[0]},resource.attributes[].{key:key,value: value.*|[0]}]|[][]`;

export const filterBySpanId = (spanId: string = '') =>
  `resourceSpans[?instrumentationLibrarySpans[?spans[?starts_with(spanId,'${spanId}')]]] | [].[instrumentationLibrarySpans[].spans[].attributes[].{key:key,value:value.*|[0],type:'span'},resource.attributes[].{key:key,value: value.*|[0],type:'resource'}]|[][]`;
