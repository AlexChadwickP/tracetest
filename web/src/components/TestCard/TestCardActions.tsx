import {Dropdown, Menu} from 'antd';
import * as S from './TestCard.styled';

interface ITestCardActions {
  testId: string;
  onDelete(): void;
}

const TestCardActions: React.FC<ITestCardActions> = ({testId, onDelete}) => {
  return (
    <Dropdown
      overlay={
        <Menu>
          <Menu.Item
            data-cy="test-delete-button"
            onClick={({domEvent}) => {
              domEvent.stopPropagation();
              onDelete();
            }}
            key="delete"
          >
            Delete
          </Menu.Item>
        </Menu>
      }
      placement="bottomLeft"
      trigger={['click']}
    >
      <span data-cy={`test-actions-button-${testId}`} className="ant-dropdown-link" onClick={e => e.stopPropagation()}>
        <S.ActionButton />
      </span>
    </Dropdown>
  );
};

export default TestCardActions;
