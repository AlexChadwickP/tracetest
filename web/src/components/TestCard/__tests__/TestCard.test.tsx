import {render} from '@testing-library/react';
import TestCard from '../TestCard';
import {TestingModels} from '../../../utils/TestingModels';

test('SpanAttributesTable', () => {
  const onDelete = jest.fn();
  const onRunTest = jest.fn();
  const onClick = jest.fn();
  const result = render(
    <TestCard onDelete={onDelete} onRunTest={onRunTest} test={TestingModels.test} onClick={onClick} />
  );
  expect(result.container).toMatchSnapshot();
});
