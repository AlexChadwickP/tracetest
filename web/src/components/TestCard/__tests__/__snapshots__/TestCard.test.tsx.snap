// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpanAttributesTable 1`] = `
<div>
  <div
    class="sc-bdvvtL CSceT"
    data-cy="test-card"
  >
    <span
      aria-label="right"
      class="anticon anticon-right"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="right"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
        />
      </svg>
    </span>
    <div
      class="sc-gsDKAQ iwigcB"
    >
      <span
        class="ant-typography sc-hKwDye gbtLIw"
      >
        
      </span>
    </div>
    <div
      class="sc-gsDKAQ iwigcB"
    >
      <span
        class="ant-typography sc-eCImPb"
      >
        GET
      </span>
    </div>
    <div
      class="sc-gsDKAQ iwigcB"
      data-cy="test-url-"
    >
      <span
        class="ant-typography sc-eCImPb"
      >
        
      </span>
    </div>
    <div
      class="sc-gsDKAQ iwigcB"
    />
    <div
      class="sc-dkPtRN eWbiec"
    >
      <button
        class="ant-btn ant-btn-primary ant-btn-background-ghost"
        data-cy="test-run-button"
        type="button"
      >
        <span>
          Run Test
        </span>
      </button>
    </div>
    <span
      class="ant-dropdown-trigger ant-dropdown-link"
      data-cy="test-actions-button-"
    >
      <span
        aria-label="more"
        class="anticon anticon-more sc-jRQBWg"
        role="img"
        style="font-size: 24px; color: rgb(154, 163, 171); cursor: pointer;"
      >
        <svg
          aria-hidden="true"
          data-icon="more"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;
